### DISCUSSION TITLE
Error while loading entity: "Cannot redefine property: name"

### DISCUSSION THREAD

**Discussion by shrugal** [mikro-orm/mikro-orm#3169] [Q&A] [answered]
**Describe the bug**
While loading an entity I get the following error: "TypeError: Cannot redefine property: name"

**Stack trace**
```
at Function.defineProperty (<anonymous>)
at /var/www/node-app/node_modules/@mikro-orm/core/entity/EntityHelper.js:96:20
at Array.forEach (<anonymous>)
at Function.defineProperties (/var/www/node-app/node_modules/@mikro-orm/core/entity/EntityHelper.js:82:14)
at Function.decorate (/var/www/node-app/node_modules/@mikro-orm/core/entity/EntityHelper.js:23:22)
at /var/www/node-app/node_modules/@mikro-orm/core/metadata/MetadataStorage.js:76:58
at Array.forEach (<anonymous>)
at MetadataStorage.decorate (/var/www/node-app/node_modules/@mikro-orm/core/metadata/MetadataStorage.js:76:14)
at MikroORM.discoverEntities (/var/www/node-app/node_modules/@mikro-orm/core/MikroORM.js:93:23)
at async Function.init (/var/www/node-app/node_modules/@mikro-orm/core/MikroORM.js:45:9)
```

**Additional context**

Compilation is done with babel, so `BABEL_DECORATORS_COMPAT` is set to `true`. While trying to find the problem I did discover that the `partnership.entity.ts` class prototype has id and name properties set to `undefined` right after resolve.

Init:

```typescript
MikroORM.init<MySqlDriver>({
    type: "mysql",
    clientUrl: url,
    user,
    password,
    metadataProvider: ReflectMetadataProvider,
    entities: [path.join(BASE_DIR, "packages/api-*/dist/*/models/*.entity.js")],
    entitiesTs: [path.join(BASE_DIR, "packages/api-*/src/*/models/*.entity.ts")],
    schemaGenerator: {
        createForeignKeyConstraints: false,
    },
    debug: process.env.NODE_ENV === "development",
});
```

Entity:

```typescript
import { Entity, Property, PrimaryKey } from "@mikro-orm/core";

@Entity()
export class Partnership {
    @PrimaryKey()
    id!: number;

    @Property()
    name: string;

    constructor(name: string) {
        this.name = name;
    }
}
```

tsconfig.json

```json
{
  "compilerOptions": {
    "target": "es6",
    "allowJs": true,
    "forceConsistentCasingInFileNames": true,
    "allowSyntheticDefaultImports": true,
    "moduleResolution": "node",
    "module": "esnext",
    "lib": ["dom", "dom.iterable", "es2020"],
    "esModuleInterop": true,
    "declaration": true,
    "composite": true,
    "baseUrl": ".",
    "paths": {},
    "typeRoots": ["node_modules/@types", "./types"],
    "skipLibCheck": true,
    "resolveJsonModule": true,
    "sourceMap": true,
    "declarationMap": true,
    "preserveConstEnums": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "strictPropertyInitialization": false,
    "useDefineForClassFields": false
  }
}
```

**Versions**

| Dependency | Version |
| - | - |
| node | 14.19.1 |
| typescript | 4.1.3 |
| babel | 7.17.6 |
| mikro-orm | 5.1.5 |
| your-driver | @mikro-orm/mysql@5.1.5 |

**Comment by B4nan** [comment]
Do you really target `es6`? Why? I don't think you should, try changin your target to something like `es2018` or higher, depends on what node version you want to use (es2018 is ok for node 14). And I would say you should also keep `useDefineForClassFields` enabled. 

edit: es2020 should be fine with node 14 too, we use that in the ORM codebase and do support node 14

Anyway, if that does not help, I will need a full reproduction for this, I don't have any experience with babel myself.

> Compilation is done with babel, so BABEL_DECORATORS_COMPAT is set to true. While trying to find the problem I did discover that the partnership.entity.ts class prototype has id and name properties set to undefined right after resolve.

I believe this is correct behaviour for babel. You might be able to control that somehow, but can't help much with that.

edit: moving to discussions, as there is no actual reproduciton. you provided quite dated tsconfig, but you say you use babel to compile, yet you did not provide any babel configuration

**Comment by shrugal** [comment, marked as answer]
Nvm, this was an error on my part. The source files are indeed compiled with babel, but during development it still uses ts-node to compile the entities. So removing `BABEL_DECORATORS_COMPAT` fixed it. Thank you for the quick answer though!
